Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java

```diff
@@ -253,25 +252,65 @@ private String getSelectedNoteLinks() {
         return links.toString();
     }
 
+    public List<Integer> getSelectedNotesPositions() {
+        SparseBooleanArray checkedPositions = getListView().getCheckedItemPositions();
+        ArrayList<Integer> positions = new ArrayList<>();
+
+        for (int i = 0; i < checkedPositions.size(); i++) {
+            if (checkedPositions.valueAt(i)) {
+                positions.add(checkedPositions.keyAt(i) - mList.getHeaderViewsCount());
+            }
+        }
+
+        return positions;
+    }
+
+    public void updateSelectionAfterTrashAction() {
+        if (DisplayUtils.isLargeScreenLandscape(getActivity())) {
+            // Try to find the nearest note to the first deleted item
+            List<Integer> deletedNotesPositions = getSelectedNotesPositions();
+            int firstDeletedNote = deletedNotesPositions.get(0);
+            int positionToSelect = -1;
+            // Loop through the notes below
+            for (int i = firstDeletedNote + 1; i < mNotesAdapter.getCount(); i++) {
+                if (!deletedNotesPositions.contains(i)) {
+                    positionToSelect = i;
+                    break;
+                }
+            }
+            if (positionToSelect == -1) {
+                // Loop through the above notes
+                for (int i = firstDeletedNote - 1; i >= 0; i--) {
+                    if (!deletedNotesPositions.contains(i)) {
+                        positionToSelect = i;
+                        break;
+                    }
+                }
+            }
+
+            if (positionToSelect != -1) {
+                Note noteToSelect = mNotesAdapter.getItem(positionToSelect + mList.getHeaderViewsCount());
+                mCallbacks.onNoteSelected(noteToSelect.getSimperiumKey(), null, noteToSelect.isMarkdownEnabled(), noteToSelect.isPreviewEnabled());
+                // As we will trigger a list refresh later, save the selectedNoteId
+                mSelectedNoteId = noteToSelect.getSimperiumKey();
+            } else {
+                // The list of notes is empty
+                if (getActivity() != null) {
+                    ((NotesActivity) requireActivity()).showDetailPlaceholder();
+                }
```

Comment:
```suggestion
                ((NotesActivity) requireActivity()).showDetailPlaceholder();
```

Do we need the `if (getActivity() != null)` check since `requireActivity()` throws an `IllegalStateException` if `getActivity() == null`?  I guess the question is do we want to do nothing or throw an `IllegalStateException` if `getActivity() == null`?  If `getActivity() == null`, that indicates something is wrong with `NoteListFragment`.  If we do nothing, it will leave the last note content in the editor and hide that something is wrong with `NoteListFragment`.  If we throw an `IllegalStateException`, it will crash and show something is wrong with `NoteListFragment`.

Reply:
As this code is called from the `TrashNotesTask`, it's wise to keep the check if the activity is not null, as it may occur after the destruction, and it won't mean that there is something wrong about `NoteListFragment`, it's just that the callback occurred after detachment, which means that we don't need to update the selected note.

Reply:
The purpose of `requireActivity()` is to throw an `IllegalStateException` if the activity is `null`.  If we're checking `getActivity() != null` just before calling `requireActivity()`, then `requireActivity()` loses its purpose.

If we're worried about `NoteListFragment` being detached when `onPostExecute` is called in `TrashNotesTask`, we can check if the fragment is detached before calling the fragment methods like this:

```java
if (!fragment.isDetached()) {
    fragment.updateSelectionAfterTrashAction();
    fragment.mActionMode.finish();
    fragment.refreshList();
}
```